### PR TITLE
gtk4: add sample/examples/application1

### PR DESCRIPTION
**Pull request by ToshioCP**
The programs corresponds to the ones in the GTK repository.

- `exampleapp.rb` is almost same as the one under `gtk3` direcotry in the Ruby-GNOME repository.
- It doesn't use `utils.rb` because of the installation. It uses `require "gtk4"` instead.
- `README.md` explains the installation and the desktop file. Its original file is `README` in the application1 directory in the original GTK repository. But some explanation is added for Ruby version.

### CODE REVIEW

File: gtk4/sample/examples/application1/README [hunk 1]
```diff
@@ -0,0 +1,25 @@
+Step 1: A trivial application
```
Comment (kou) [inline gtk4/sample/examples/application1/README:1]: How about renaming this file to `README.md` and use Markdown?

File: gtk4/sample/examples/application1/exampleapp.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
+exec ruby -x "$0" "$@"
+#!ruby
+
```
Comment (kou) [inline gtk4/sample/examples/application1/exampleapp.rb:4]: How about simplifying this shebang?

```suggestion
#!/usr/bin/env ruby
```

File: gtk4/sample/examples/application1/exampleapp.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+#
+# Copyright (C) 2022  Ruby-GNOME Project Team
```
Comment (kou) [inline gtk4/sample/examples/application1/exampleapp.rb:6]:
```suggestion
# Copyright (C) 2023  Ruby-GNOME Project Team
```

File: gtk4/sample/examples/application1/exampleapp.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+#
+# License: LGPL2.1+
+
+require "gtk4"
+
+class ExampleAppWindow < Gtk::ApplicationWindow
+
+  def open(file)
+    
+  end
```
Comment (kou) [inline gtk4/sample/examples/application1/exampleapp.rb:32]:
```suggestion
  def open(file)
  end
```

File: gtk4/sample/examples/application1/exampleapp.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+# example from https://gitlab.gnome.org/GNOME/gtk/blob/main/examples/window-default.c
```
Comment (kou) [inline gtk4/sample/examples/application1/exampleapp.rb:22]: Is this correct?

```suggestion
# Example from:
# * https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application1/exampleapp.c
# * https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application1/exampleappwin.c
```

File: gtk4/sample/examples/application1/exampleapp.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      win = nil
+      unless windows.empty?
+        win = windows.first
+      else
+        win = ExampleAppWindow.new(application)
+      end
+
+      files.each { |file| win.open(file) }
+        
+      win.present
```
Comment (kou) [inline gtk4/sample/examples/application1/exampleapp.rb:54]: How about simplifying this?

```suggestion
      window = application.windows[0] || ExampleAppWindow.new(application)
      files.each do |file|
        window.open(file)
      end
      window.present
```

File: gtk4/sample/examples/application1/exampleapp.rb [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    end
+  end
+end
+
+app = ExampleApp.new
+
+puts app.run
```
Comment (kou) [inline gtk4/sample/examples/application1/exampleapp.rb:61]: Do we need to `puts` here?

```suggestion
app = ExampleApp.new
app.run
```

File: gtk4/sample/examples/application1/README [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+To make gnome-shell use the desktop file and icon for this example
+while running it uninstalled, do the following:
+
+mkdir -p ~/.local/share/applications
+sed -e "s#@bindir@#$PWD#" org.gtk.exampleapp.desktop \
+  > ~/.local/share/applications/org.gtk.exampleapp.desktop
+
+mkdir -p ~/.local/share/icons/hicolor/48x48/apps
+cp exampleapp.png ~/.local/share/icons/hicolor/48x48/apps
```
Comment (kou) [inline gtk4/sample/examples/application1/README:11]: Could you also mention the source of `exampleapp.png` ( https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application1/exampleapp.png ) and its license in this file?

**Review by kou — COMMENTED** [review]
Thanks!

**Comment by ToshioCP** [comment]
Thank your for the suggestion.

I've corrected mistakes you had pointed and added a bit to the README.md.
I've checked all the files but it might have some mistakes because of  my carelessness.

Toshio Sekiya

File: gtk4/sample/examples/application1/README.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
+# Step 1: A trivial application
+
+The following is only useful in the desktop system based on [Desktop Entry Specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html) such as Gnome.
```
Comment (kou) [inline gtk4/sample/examples/application1/README.md:3]:
```suggestion
The following is only useful in the desktop system based on [Desktop Entry Specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html) such as GNOME.
```

File: gtk4/sample/examples/application1/README.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+To make gnome-shell use the desktop file and icon for this example while running it uninstalled, do the following:
+
+```
```
Comment (kou) [inline gtk4/sample/examples/application1/README.md:7]:
```suggestion
```console
```

File: gtk4/sample/examples/application1/README.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+$ mkdir -p ~/.local/share/applications
+$ cp org.gtk.exampleapp.desktop ~/.local/share/applications/org.gtk.exampleapp.desktop
+$ mkdir -p ~/.local/share/icons/hicolor/48x48/apps
+$ cp exampleapp.png ~/.local/share/icons/hicolor/48x48/apps
+```
+
+The file exapmleapp.png is from the GTK GitLab repository.
+The original file is [here](https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application1) and the license is LGPL.
```
Comment (kou) [inline gtk4/sample/examples/application1/README.md:15]:
```suggestion
The original file is [here](https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application1/exampleapp.png) and the license is LGPL 2.1 or later.
```

File: gtk4/sample/examples/application1/README.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+Note:
+
+1. You need to install your exampleapp.rb in advance.
+
+        $ cp exampleapp.rb $HOME/.local/bin/exampleapp.rb
+        $ chmod 755 $HOME/.local/bin/exampleapp.rb
```
Comment (kou) [inline gtk4/sample/examples/application1/README.md:22]:
```suggestion
   ```console
   $ cp exampleapp.rb ~/.local/bin/exampleapp.rb
   $ chmod +x ~/.local/bin/exampleapp.rb
   ```
```

File: gtk4/sample/examples/application1/README.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+
+2. If you use rbenv, your ruby executable may not be found.
+You can solve this by creating a symbolic link in $HOME/.local/bin to your ruby executable.
+For example,
```
Comment (kou) [inline gtk4/sample/examples/application1/README.md:26]:
```suggestion
   You can solve this by creating a symbolic link in `~/.local/bin` to your ruby executable.
   For example,
```

File: gtk4/sample/examples/application1/README.md [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+
+        $ cd $HOME/.local/bin
+        $ ln -s $HOME/.rbenv/versions/3.1.2/bin/ruby ruby
```
Comment (kou) [inline gtk4/sample/examples/application1/README.md:29]:
```suggestion
   ```console
   $ cd ~/.local/bin
   $ ln -s $(rbenv which ruby) ruby
   ```
```

File: gtk4/sample/examples/application1/exampleapp.rb [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+# Example from:
```
Comment (kou) [inline gtk4/sample/examples/application1/exampleapp.rb:18]:
```suggestion
#
# Example from:
```

File: gtk4/sample/examples/application1/exampleapp.rb [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+# * https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application1/exampleapp.c
+# * https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application1/exampleappwin.c#
+# License: LGPL2.1+
```
Comment (kou) [inline gtk4/sample/examples/application1/exampleapp.rb:21]: How about using SPDX identifier?
https://spdx.org/licenses/

```suggestion
# License: LGPL-2.1-or-later
```

File: gtk4/sample/examples/application1/exampleapp.rb [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+      files.each do |file|
+        window.open(file)
+      end
+      window.present
+    end
+  end
+end
+
+app = ExampleApp.new
+app.run
```
Comment (kou) [inline gtk4/sample/examples/application1/exampleapp.rb:49]: Could you add a new line here?

**Comment by ToshioCP** [comment]
Two files README.md and exampleapp.rb have been modified along your suggestion.

Toshio Sekiya

File: gtk4/sample/examples/application1/README.md [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+The original file is [here](https://gitlab.gnome.org/GNOME/gtk/-/blob/main/examples/application1/exampleapp.png) and the license is LGPL 2.1 or later.
+
+Note:
+
+1. You need to install your exampleapp.rb in advance.
+
+```console
+$ cp exampleapp.rb $HOME/.local/bin/exampleapp.rb
+$ chmod 755 $HOME/.local/bin/exampleapp.rb
+```
```
Comment (kou) [inline gtk4/sample/examples/application1/README.md:24]:
```suggestion
   ```console
   $ cp exampleapp.rb ~/.local/bin/exampleapp.rb
   $ chmod +c ~/.local/bin/exampleapp.rb
   ```
```

File: gtk4/sample/examples/application1/README.md [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+
+2. If you use rbenv, your ruby executable may not be found.
+   You can solve this by creating a symbolic link in `~/.local/bin` to your ruby executable.
+   For example,
+
+```console
+$ cd $HOME/.local/bin
+$ ln -s $HOME/.rbenv/versions/3.1.2/bin/ruby ruby
+```
```
Comment (kou) [inline gtk4/sample/examples/application1/README.md:33]:
```suggestion
   ```console
   $ ln -s $(rbenv which ruby) ~/.local/bin/ruby
   ```
```

**Comment by kou** [comment]
Thanks!
I've merged with some minor style adjustments.

**Pull request merged**